src/SharedPipes/CompileClientDataJson.php

```diff
@@ -32,24 +34,29 @@ public function handle(AssertionValidation|AttestationValidation $validation, Cl
     {
         try {
             $object = json_decode(
-                base64_decode($validation->request->json('response.clientDataJSON', '')), false, 32, JSON_THROW_ON_ERROR
+                base64_decode($validation->request->json('response.clientDataJSON', '')),
+                false,
+                32,
+                JSON_THROW_ON_ERROR
             );
-        } catch (JsonException) {
+        } catch (JsonException|UrlException) {
             static::throw($validation, 'Client Data JSON is invalid or malformed.');
         }
 
-        if (!$object) {
+        if ($object === '' || $object === null || $object === []) {
             static::throw($validation, 'Client Data JSON is empty.');
         }
 
         foreach (['type', 'origin', 'challenge'] as $key) {
-            if (!isset($object->{$key})) {
-                static::throw($validation, "Client Data JSON does not contain the [$key] key.");
+            if (!isset($object->{$key})) { // @phpstan-ignore-line
```

Comment:
```suggestion
            if (!property_exists($object, $key)) {
```

should help to avoid `@phpstan-ignore-line`